### PR TITLE
Simplify restore logic.

### PR DESCRIPTION
**Pull request by obino**
With the previous work to make some start-up functions idempotent, we can
simplify the logic used to restore the  AppController state.

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1016/

File: AppController/djinn.rb [hunk 1]
```diff
@@ -2693,56 +2692,38 @@ def backup_appcontroller_state()
...
-
   # Restores the state of each of the instance variables that the AppController
   # holds by pulling it from ZooKeeper (previously populated by the Shadow
   # node, who always has the most up-to-date version of this data).
   #
   # Returns:
-  #   Two booleans, that indicate if (1) data was restored to this AppController
-  #   from either ZooKeeper or locally, and (2) if we need to start the roles
-  #   on this machine or not.
+  #   a boolean to indicate if we were able to restore the state from
```
Comment (menivaitsi) [inline AppController/djinn.rb]: a boolean -> A boolean

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1048/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/1050/

**Pull request merged**